IdentityCore/src/requests/broker/MSIDSSOXpcSilentTokenRequest.m

```diff
@@ -0,0 +1,108 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import "MSIDSSOXpcSilentTokenRequest.h"
+#import "MSIDSSOExtensionRequestDelegate.h"
+#import "MSIDBrokerOperationSilentTokenRequest.h"
+#import "MSIDRequestParameters.h"
+#import "MSIDXpcSingleSignOnProvider.h"
+#import "MSIDBrokerOperationTokenResponse.h"
+
+@interface MSIDSSOXpcSilentTokenRequest()
+
+@property (nonatomic, copy) MSIDSSOExtensionRequestDelegateCompletionBlock completionBlock;
+@property (nonatomic, copy) MSIDRequestCompletionBlock requestCompletionBlock;
+@property (nonatomic) MSIDBrokerOperationSilentTokenRequest *operationRequest;
+@property (nonatomic) MSIDXpcSingleSignOnProvider *xpcSingleSignOnProvider;
+@property (nonatomic) id<MSIDRequestContext> context;
+
+@end
+
+@implementation MSIDSSOXpcSilentTokenRequest
+
+@synthesize requestCompletionBlock, operationRequest;
+
+- (instancetype)initWithRequestParameters:(MSIDRequestParameters *)parameters
+                             forceRefresh:(BOOL)forceRefresh
+                             oauthFactory:(MSIDOauth2Factory *)oauthFactory
+                   tokenResponseValidator:(MSIDTokenResponseValidator *)tokenResponseValidator
+                               tokenCache:(id<MSIDCacheAccessor>)tokenCache
+                     accountMetadataCache:(MSIDAccountMetadataCacheAccessor *)accountMetadataCache
+                       extendedTokenCache:(nullable id<MSIDExtendedTokenCacheDataSource>)extendedTokenCache
+{
+    self = [super initWithRequestParameters:parameters
+                               forceRefresh:forceRefresh
+                               oauthFactory:oauthFactory
+                     tokenResponseValidator:tokenResponseValidator
+                                 tokenCache:tokenCache
+                       accountMetadataCache:accountMetadataCache
+                         extendedTokenCache:extendedTokenCache];
+    if (self)
+    {
+        self.completionBlock = [super getCompletionBlock];
+        self.xpcSingleSignOnProvider = [MSIDXpcSingleSignOnProvider new];
+        self.context = parameters;
+    }
+
+    return self;
+}
+
+- (void)executeRequestImplWithCompletionBlock:(MSIDRequestCompletionBlock _Nonnull)completionBlock
+{
+    NSDictionary *jsonDictionary = [self.operationRequest jsonDictionary];
+
+    if (!jsonDictionary)
+    {
+        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"Failed to serialize SSO request dictionary for silent token request", nil, nil, nil, self.requestParameters.correlationId, nil, YES);
+        completionBlock(nil, error);
+        return;
+    }
+
+    [self performXpcRequest:jsonDictionary];
+    self.requestCompletionBlock = completionBlock;
```

Comment:
Will it be safer to move line 84 before line 83? Same pattern can be found in `MSIDSSOExtensionSilentTokenRequest.m`

Reply:
Yes, and Moved